docs/advanced/additional-requests.rst

```diff
@@ -0,0 +1,503 @@
+
+.. code-block:: python
+
+    import attrs
+    import web_poet
+
+    def request_implementation(req: web_poet.HttpRequest) -> web_poet.HttpResponse:
+        ...
+
+
+    def create_http_client():
+        return web_poet.HttpClient()
+
+
+    @attrs.define
+    class SomePage(web_poet.ItemWebPage):
+        http_client: web_poet.HttpClient
+
+        async def to_item(self):
+            ...
+
+    # Once this is set, the ``request_implementation`` will become available to
+    # all instances of HttpClient unless a ``request_downloader`` is injected
+    # to it (see #2 Dependency Injection example below).
+    web_poet.request_backend_var.set(request_implementation)
+
+    # Assume that it's constructed with the necessary arguments taken somewhere.
+    response = web_poet.HttpResponse(...)
+
+    page = SomePage(response=response, http_client=create_http_client())
+    item = page.to_item()
+
+Setting this up would allow access to the request implementation in a
+:class:`~.HttpClient` instance which uses it by default.
+
+.. warning::
+
+    If no value for ``web_poet.request_backend_var`` was set, then a
+    :class:`~.RequestBackendError` is raised. However, no exception would
+    be raised if **option 2** below is used.
+
+
+2. Dependency Injection
+-----------------------
+
+The framework using **web-poet** might be using other libraries which doesn't
+have a full support to :mod:`contextvars` `(e.g. Twisted)`. With that, an
+alternative approach would be to supply the request implementation when creating
+an :class:`~.HttpClient` instance:
+
+.. code-block:: python
+
+    import attrs
+    import web_poet
+
+    def request_implementation(req: web_poet.HttpRequest) -> web_poet.HttpResponse:
+        ...
+
+    def create_http_client():
+        return web_poet.HttpClient(request_downloader=request_implementation)
+
+
+    @attrs.define
+    class SomePage(web_poet.ItemWebPage):
+        http_client: web_poet.HttpClient
+
+        async def to_item(self):
+            ...
+
+    # Assume that it's constructed with the necessary arguments taken somewhere.
+    response = web_poet.HttpResponse(...)
+
+    page = SomePage(response=response, http_client=create_http_client())
+    item = page.to_item()
+
+From the code sample above, we can see that every time an :class:`~.HttpClient`
+is created for Page Objects needing an ``http_client``, the specific **request
+implementation** from a given framework is injected to it.
+
+
+Exception Handling
+==================
+
+All exceptions that the HTTP Downloader Implementation explicitly raises when
+implementing it for **web-poet** should be :class:`web_poet.exceptions.http.HttpRequestError`
+or a subclass from it.
+
+For frameworks that implement and use **web-poet**, exceptions that ocurred when
+handling the additional requests like `connection errors`, `time outs`, `TLS
+errors`, etc should be replaced by :class:`web_poet.exceptions.http.HttpRequestError`
+by raising it explicitly.
```

Comment:
I think we should document explicitly (here and in HttpRequestError docstring) that this error shouldn't be raised when the response is received, but the status code is not 200. Or that it should be raised :) We should have the behavior clear.

Reply:
Addressed this in https://github.com/scrapinghub/web-poet/pull/22/commits/7947d588c35edc9e6498480d36281e1d93b8ba2e